utils/api.py

```diff
@@ -459,11 +459,15 @@ def __init__(self, obj):
 
     def reload(self):
         self._obj.reload_if_needed()
+        reloaded_actions = []
         for action in self._obj._actions:
-            setattr(
-                self,
-                action["name"],
-                Action(self, action["name"], action["method"], action["href"]))
+            # don't redefine actions with same name and different methods
```

Comment:
There often are, e.g. delete action with POST and delete action with DELETE. The code used to override (redefine) the first one (POST), so we were using DELETE where it was available and POST where it wasn't. I think it's better to have consistent behavior and always use the first method (i.e. POST) by default. I added "force_method" option so it's possible to choose non-default method and to test all methods that are available.